### PR TITLE
[PHP-1] add classes Basics and BasicsValidator

### PR DESCRIPTION
**Pull request by oleksiivv**

### CODE REVIEW

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]
Please rewrite code based on php code standarts.
Please make sure unit tests are passing.
Why do you need try/catch here ?

File: src/basics/Basics.php [hunk 1]
```diff
@@ -0,0 +1,105 @@
...
+     * @param int $minute
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    public function getMinuteQuarter(int $minute): string
+    {
+        $this->baseValidator->isMinutesException($minute);
+
+        $result="";
+        if(
```
Comment (yderevianko-cartera) [inline src/basics/Basics.php:32]: `  if ($minute>0 && $minute<=15) {
       $result=self::HOUR_QUARTERS[0];
   } elseif ($minute>15 && $minute<=30) {
`

File: src/basics/Basics.php [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+     * @throws \InvalidArgumentException
+     */
+    public function isSumEqual(string $input): bool
+    {
+        $this->baseValidator->isValidStringException($input);
+        
+        $sumFirstNums=0;
+        $sumLastNums=0;
+
+        for($i=0;$i<strlen($input)/2;$i++){
```
Comment (yderevianko-cartera) [inline src/basics/Basics.php:95]: please read about php code style

**Review by yderevianko-cartera — CHANGES_REQUESTED** [review]

File: src/basics/Basics.php [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+     * @return string
+     * @throws \InvalidArgumentException
+     */
+    public function getMinuteQuarter(int $minute): string
+    {
+        $this->baseValidator->isMinutesException($minute);
+
+        $result = "";
+        if ($minute > 0 && $minute <= 15) {
+
```
Comment (yderevianko-cartera) [inline src/basics/Basics.php:33]: why do you need so many empty lines ?

File: src/basics/Basics.php [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+     * @throws \InvalidArgumentException
+     */
+    public function isSumEqual(string $input): bool
+    {
+        $this->baseValidator->isValidStringException($input);
+        
+        $sumOfFirstPart = 0;
+        $sumOfSecondPart = 0;
+
+        for($i = 0; $i < strlen($input) / 2; $i++) {
```
Comment (yderevianko-cartera) [inline src/basics/Basics.php:86]: missed space after for

File: src/basics/Basics.php [hunk 5]
```diff
@@ -0,0 +1,96 @@
...
+     * @param int $year
+     * @return boolean
+     * @throws \InvalidArgumentException
+     */
+    public function isLeapYear(int $year): bool
+    {
+        
+        $this->baseValidator->isYearException($year);
+        
+        if ((
```
Comment (yderevianko-cartera) [inline src/basics/Basics.php:63]: can be changed to single line ternar if 
return (( $year % 4 === 0 && $year % 100 !== 0) || $year % 400 === 0 ))

Comment (oleksiivv) [reply src/basics/Basics.php]: done

File: src/basics/Basics.php (hunk 4 above)
Comment (oleksiivv) [reply src/basics/Basics.php]: done

File: src/basics/Basics.php (hunk 3 above)
Comment (oleksiivv) [reply src/basics/Basics.php]: fixed

**Review by yderevianko-cartera — APPROVED** [review]

**Pull request merged**